contract.md

```diff
@@ -12,29 +14,48 @@ like to work on, as part of your project (for example, other things related to D
 you have come across, or topics from other classes)? Why are you interested in working on
 these things?_
 
+	First and foremost, I want to work on the skills used for external DSLs more than I want to work on the skills needed for an internal DSL. This is because external DSLs would work on skills that are more unique to programming language development- technically you have to come up with a way to parse text, and design-wise you're free to design a syntax from the ground up, rather than just working around the the host language's specific syntax. 
+
+	Besides that, I'd also like to touch on the idea of 
+
 ## Time management plan
 
 _How do you plan to set aside time outside of class, to work on the project? Are there
 intermediate milestones that you can create, to help you make consistent progress?_
 
+	I'd like to work on the project at a consistent rate, putting in a specific number of hours a week towards designing the project. I also want to work on it in specific blocks of time dedicated ot the project, rather than just working on it haphazardly.
+	The project can be divided into a few key phases. The first one would be frontend design - coming up with the syntax and the final product, and figuring out what libraries I might need to work with for stuff like editing images or sound files. Next would be coming up with, and then implementing, a structure for the backend - A set of functions that would be able to build up to the wanted output. After that would be a parser that can convert from the external DSL to these functions, and the final phase would be polishing what was already done to make the language more usable based on other's feedback. 
```

Comment:
This sounds solid! I like the idea of dividing it into phases. Would each phase take a different amount of time?